langchain4j-core/src/main/java/dev/langchain4j/model/chat/observability/ChatLanguageModelListener.java

```diff
@@ -0,0 +1,48 @@
+package dev.langchain4j.model.chat.observability;
+
+import dev.langchain4j.Experimental;
+
+/**
+ * TODO
+ */
+// TODO name: ChatLanguageModelObserver?
+// TODO package
+@Experimental
+public interface ChatLanguageModelListener {
+
+
+    /**
+     * TODO
+     *
+     * @param request
+     */
+    // TODO names
+    @Experimental
+    default void onRequest(ChatLanguageModelRequest request) {
+
+    }
+
+    /**
+     * TODO
+     *
+     * @param request
+     * @param response
+     */
+    // TODO names
+    @Experimental
+    default void onResponse(ChatLanguageModelRequest request, ChatLanguageModelResponse response) {
+
+    }
+
+    /**
+     * TODO
+     *
+     * @param request
+     * @param error
+     */
+    // TODO names
+    @Experimental
+    default void onError(ChatLanguageModelRequest request, Throwable error) {
```

Comment:
I think we should follow the same approach as the vert.x tracing API and squash `onResponse` and the `onError` methods.

Reply:
I would still prefer to differentiate between "successful response" and "error" on the API level, I think it makes more sense from the user perspective.

Given the following API:
```java
public interface ModelListener<Request, Response> {

    default void onRequest(Request request) {

    }

    default void onResponse(Request request, Response response) {

    }

    default void onError(Request request, Response response, Throwable error) {

    }
}
```

Given that "squashing" will be required for tracing, it still can be done on the implementor side like this:
```java
ModelListener modelListener = new ModelListener() {

    @Override
    public void onRequest(Request request) {
        // process request
    }

    @Override
    public void onResponse(Request request, Response response) {
        onError(request, response, null);
    }

    @Override
    public void onError(Request request, Response response, Throwable error)    {
        // process response and error
    }
};
```

Would that work for you?

Reply:
No strong feeling here. This should be fine:
```
public interface ModelListener<Request, Response> {

    default void onRequest(Request request) {

    }

    default void onResponse(Request request, Response response) {

    }

    default void onError(Request request, Response response, Throwable error) {

    }
}
```